### DISCUSSION TITLE
500 Internal Server Error for vercel edge runtime

### DISCUSSION THREAD

**Discussion by aplr** [vercel/community#1625] [Help]
### Summary

Hi there,

I have a nextjs app built 100% on top of the app directory. Everything's working just fine until switching to the edge runtime by setting `experimental.runtime` to `experimental-edge`. Here's the configurations I've tried:

* ✅ Running the app _locally_ with runtime `nodejs` (`next start`)
* ✅ Running the app _locally_ with runtime `experimental-edge` (`next start`)
* ✅ Running the app _on vercel_ with runtime `nodejs`
* 🆘 Running the app _on vercel_ with runtime `experimental-edge`

~I'm a bit lost, as the function logs are empty and I have no way of finding out what's going on really.~

I've found some logs on axiom:

```json
{
  "request": {
    "host": "app-49ltp2rre-org.vercel.app",
    "id": "fra1::fra1::q9nbg-1677361775461-1092f367a262",
    "ip": "1.2.3.4",
    "method": "HEAD",
    "path": "/_next/data/qDssMCIjEfFqDPbA9ajoq/index.json",
    "scheme": "https",
    "statusCode": 500,
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/111.0"
  },
  "vercel": {
    "deploymentId": "dpl_H1Hi8vZ5Dnhis3aKRuSCef5H94n7",
    "deploymentURL": "app-76i0q15bn-org.vercel.app",
    "projectId": "prj_LpZ4uWzLziwFxqnyycm0YanNRgjA",
    "projectName": "app",
    "region": "fra1",
    "route": "/_next/data/qDssMCIjEfFqDPbA9ajoq/index.json",
    "source": "lambda",
    "teamName": "org"
  }
}
```

The weird thing here is that the id in the request host `49ltp2rre` does not match the id in the deploymentUrl `76i0q15bn`. The last one comes from another deployment on different branch.

Did you experience something similar, or do you have any ideas what I could try?

### Additional information

`next@13.2.1`

**Comment by amyegan** [comment]
Thanks for reaching out, Andreas. Are you able share the error message or some more details about what is going wrong? A minimal repro would also be very helpful for debugging if you're able to share one
